src/addons/TransitionablePortal/TransitionablePortal.js

```diff
@@ -99,13 +97,8 @@ export default class TransitionablePortal extends Component {
 
   handlePortalClose = () => {
     debug('handlePortalClose()')
-    const { open } = this.props
-    const { portalOpen } = this.state
 
-    // Heads up! We simply call `onClose` when component is controlled with `open` prop.
-    // But, when it's autocontrolled we should change the state to opposite to keep the transition
-    // queue
-    if (_.isNil(open)) this.setState({ portalOpen: !portalOpen })
+    this.setState({ portalOpen: false })
```

Comment:
I rethought the state handling there and as I see, it was overcomplicated. I've simplified it and it looks that everything is correct.